src/mesido/workflows/gas_elect_workflow.py

```diff
@@ -0,0 +1,244 @@
+import logging
+import os
+
+from mesido.esdl.esdl_additional_vars_mixin import ESDLAdditionalVarsMixin
+from mesido.esdl.esdl_mixin import ESDLMixin
+from mesido.head_loss_class import HeadLossOption
+from mesido.techno_economic_mixin import TechnoEconomicMixin
+from mesido.workflows.goals.minimize_tco_goal import MinimizeTCO
+from mesido.workflows.io.write_output import ScenarioOutput
+from mesido.workflows.utils.helpers import main_decorator
+
+import numpy as np
+
+from rtctools.optimization.collocated_integrated_optimization_problem import (
+    CollocatedIntegratedOptimizationProblem,
+)
+from rtctools.optimization.goal_programming_mixin import Goal
+from rtctools.optimization.linearized_order_goal_programming_mixin import (
+    LinearizedOrderGoalProgrammingMixin,
+)
+from rtctools.optimization.single_pass_goal_programming_mixin import (
+    CachingQPSol,
+    SinglePassGoalProgrammingMixin,
+)
+from rtctools.util import run_optimization_problem
+
+logger = logging.getLogger("WarmingUP-MPC")
+logger.setLevel(logging.INFO)
+
+
+class TargetHeatGoal(Goal):
+    priority = 1
+
+    order = 2
+
+    def __init__(self, state, target):
+        self.state = state
+
+        self.target_min = target
+        self.target_max = target
+        try:
+            self.function_range = (-1.0e6, max(2.0 * max(target.values), 1.0e6))
+            self.function_nominal = max(np.median(target.values), 1.0e6)
+        except Exception:
+            self.function_range = (-1.0e6, max(2.0 * target, 1.0e6))
+            self.function_nominal = max(target, 1.0e6)
+
+    def function(self, optimization_problem, ensemble_member):
+        return optimization_problem.state(self.state)
+
+
+# class MinimizeSourcesHeatGoal(Goal):
+#     """
+#     A minimization goal for source milp production. We use order 1 here as we want to minimize milp
+#     over the full horizon and not per time-step.
+#     """
+
+#     priority = 3
+
+#     order = 1
+
+#     def __init__(self, source: str):
+#         """
+#         The constructor of the goal.
+
+#         Parameters
+#         ----------
+#         source : string of the source name that is going to be minimized
+#         """
+#         self.source = source
+
+#     def function(
+#         self, optimization_problem: CollocatedIntegratedOptimizationProblem, ensemble_member: int
+#     ) -> ca.MX:
+#         """
+#         This function returns the state variable to which should to be matched to the target
+#         specified in the __init__.
+
+#         Parameters
+#         ----------
+#         optimization_problem : The optimization class containing the variables'.
+#         ensemble_member : the ensemble member.
+
+#         Returns
+#         -------
+#         The Heat_source state of the optimization problem.
+#         """
+#         return optimization_problem.state(f"{self.source}.Gas_source_mass_flow")
```

Comment:
Do you plan to use it, otherwise remove it.

Reply:
Shucks. I forgot to delete this. Thanks, it is deleted now